### PR TITLE
[esbmc] updated ESBMC into version 6.10.0

### LINKED ISSUE #724: ESBMC 6.10
We should prepare for the next release of ESBMC. @fbrausse has solved lots of union problems recently.

So, which open PRs should we wait for the next release? Is there any other features that we should add before it?

### PR DESCRIPTION
**Pull request by rafaelsamenezes**
Should close #724 

Let me know if I missed something on the release notes.

### CODE REVIEW

**Comment by mikhailramalho** [comment]
Do you want the atexit to be part of release? @lucasccordeiro

Otherwise, I say we are good to go.

File: scripts/release-notes.txt [hunk 1]
```diff
@@ -1,5 +1,14 @@
 Release Notes of the ESBMC model checker
 
+*** Version 6.10
+* Lots of bugs cases relating unions have been fixed
```
Comment (fbrausse) [inline scripts/release-notes.txt]: english b0rked. "Lots of bugs related to unions have been fixed".

File: scripts/release-notes.txt [hunk 2]
```diff
@@ -1,5 +1,14 @@
...
+* Compound Literals are now supported
+*  _ExtInt struct fields now have proper padding and alignment
```
Comment (fbrausse) [inline scripts/release-notes.txt]: Remove one of the two spaces at the beginning of this line.

File: scripts/release-notes.txt [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
+* A bug where the size of 3-dimensional arrays was being not being computed correctly has been solved
+* A new intrinsic for stack allocation has been added: __ESBMC_alloca
+* The Solidity frontend now has support for: bool literal and while
+* New options have been added: file-output and cex-output
```
Comment (fbrausse) [inline scripts/release-notes.txt]: please include the `--` to make it clear what exactly the options are

**Comment by lucasccordeiro** [comment]
> Do you want the atexit to be part of release? @lucasccordeiro

Do you think you can finish the `atexit` PR before you leave on Wednesday? If so, I would recommend waiting for your PR before we release v6.10.

@fbrausse and @rafaelsamenezes: do you have any thoughts here?

**Review by lucasccordeiro — CHANGES_REQUESTED** [review]
@rafaelsamenezes: I agree with all comments from @fbrausse.

In addition to that, can I also ask you to update our documentation concerning the `__ESBMC_alloca`?

See https://ssvlab.github.io/esbmc/documentation.html.

**Comment by lucasccordeiro** [comment]
@rafaelsamenezes: can you also please mention the contractors implemented by @Mohannad-Aldughaim? You can cite this technical report: https://arxiv.org/pdf/2012.11245.pdf

File: scripts/release-notes.txt (hunk 1 above)
Comment (rafaelsamenezes) [reply scripts/release-notes.txt]: Done

File: scripts/release-notes.txt (hunk 2 above)
Comment (rafaelsamenezes) [reply scripts/release-notes.txt]: Done

File: scripts/release-notes.txt (hunk 3 above)
Comment (rafaelsamenezes) [reply scripts/release-notes.txt]: Done

File: scripts/release-notes.txt [hunk 4]
```diff
@@ -1,5 +1,15 @@
 Release Notes of the ESBMC model checker
 
+*** Version 6.10
+* Lots of bugs related to unions have been fixed"
```
Comment (fbrausse) [inline scripts/release-notes.txt]: There is a quote character at the end.

File: scripts/release-notes.txt [hunk 5]
```diff
@@ -1,5 +1,15 @@
...
+* Compound Literals are now supported
+* _ExtInt struct fields now have proper padding and alignment
+* A bug where the size of 3-dimensional arrays was being not being computed correctly has been solved
```
Comment (fbrausse) [inline scripts/release-notes.txt]: Please remove the first "being" here.

**Review by fbrausse — COMMENTED** [review]
Sorry, two more tiny changes; otherwise LGTM.

**Comment by lucasccordeiro** [comment]
@rafaelsamenezes: we should advertise this release on LinkedIn as you did last time.

File: scripts/release-notes.txt (hunk 4 above)
Comment (rafaelsamenezes) [reply scripts/release-notes.txt]: Done

File: scripts/release-notes.txt (hunk 5 above)
Comment (rafaelsamenezes) [reply scripts/release-notes.txt]: Done

**Review by fbrausse — APPROVED** [review]

**Review by lucasccordeiro — APPROVED** [review]
LGTM.

**Comment by lucasccordeiro** [comment]
@rafaelsamenezes: we need to mention the fixes from @kunjsong01 concerning our Clang C++ frontend.

**Comment by lucasccordeiro** [comment]
@kunjsong01: could you please contribute a few sentences about your fixes to our release notes?

File: scripts/release-notes.txt [hunk 6]
```diff
@@ -1,5 +1,15 @@
 Release Notes of the ESBMC model checker
 
+*** Version 6.10
+* Lots of bugs related to unions have been fixed
+* Compound Literals are now supported
+* _ExtInt struct fields now have proper padding and alignment
+* A bug where the size of 3-dimensional arrays was not being computed correctly has been solved
+* A new intrinsic for stack allocation has been added: __ESBMC_alloca
+* The Solidity frontend now has support for: bool literal and while
+* New options have been added: --file-output, --cex-output, --initialize-nondet-variables, and --non-supported-models-as-zero
```
Comment (kunjsong01) [inline scripts/release-notes.txt]: * The clang-cpp-frontend now has support for: CXXThisExpr and CXXConstructor.

**Comment by kunjsong01** [comment]
> @kunjsong01: could you please contribute a few sentences about your fixes to our release notes?

Done. Added as review comment.

**Pull request merged**